File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
// Testcase 16번만 못뚫었음 ㅇㅅㅇ ...
// reportInfo에는 key(본인)을 신고한 사람의 리스트 저장
// result에는 내가 신고한사람이 몇명 정지당했는지 저장
function solution(id_list, report, k){
  let reportInfo = {};
  let result = {};
  id_list.forEach(userId => {
    reportInfo[userId] = [];
    result[userId] = 0;
  });

  new Set(report).forEach((reportStr) => {
    let [userId, targetId] = reportStr.split(" ");

    if(!(reportInfo[targetId].includes(userId))) reportInfo[targetId].push(userId);
  });

  Object.entries(reportInfo).forEach(([_, reporters]) => {
    if(reporters.length >= k){
      reporters.forEach(user => result[user]++);
    }
  })

  return Object.values(result);
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = {key: [] for key in id_list}
    result = {key: 0 for key in id_list}
    for info in report:
        userId, targetId = info.split(" ")
        if userId not in answer[targetId]:
            answer[targetId].append(userId)
    
    for userId in answer:
        if len(answer[userId]) >= k:
            for user in answer[userId]:
                result[user] += 1
    return list(result.values())